src/main/kotlin/Actions/CreateToDoAction.kt

```diff
@@ -1,23 +1,44 @@
 package Actions
 
-import Todo.ToDo
+import Errors.BadRequestError
+import Errors.UnsupportedMediaError
+import Todo.ToDoList
 import contenttype.HttpContentTypes
 import httpstatus.HttpStatus
+import request.Request
 import response.ResponseBuilder
 
-class CreateToDoAction(private val toDo: ToDo): Action {
+class CreateToDoAction(private val toDoList: ToDoList): Action {
     private val headers = mapOf("Content-Type" to (HttpContentTypes.JSON.type + HttpContentTypes.JSON.parameter))
     private lateinit var requestBody: String
 
-    override fun act(responseBuilder: ResponseBuilder): String {
-        val responseBody = toDo.createToDo(requestBody)
-        responseBuilder.setBody(responseBody)
-        responseBuilder.setHeaders(headers)
-        responseBuilder.setStatusCode(HttpStatus.Created)
-        return responseBuilder.build()
+    override fun act(responseBuilder: ResponseBuilder, request: Request): String {
+        return if (validContentType(request) && validBody(request) && toDoList.addTask(requestBody)) {
+            responseBuilder.setBody(requestBody)
+            responseBuilder.setHeaders(headers)
+            responseBuilder.setStatusCode(HttpStatus.Created)
+            responseBuilder.build()
+        } else if (validContentType(request) && (!validBody(request) || !toDoList.addTask(requestBody))) {
```

Comment:
I'm not a big fan of `toDoList.addTask(requestBody)` being run twice in one call (lines 16 and 21). Looks like `toDoList.addTask` isn't a pure function, so you could get a different result on the second call.

You could probably save the response in a variable, so you don't have to call the function multiple times.